### PR TITLE
Datacache correction

### PR DESCRIPTION
**Pull request by Qiao-Jin**
This PR fixes Datacache commit strategy, and is the precondition of PR #1602
But it seems that this PR will give rise to some error in UT_Wallet, @shargon  could you please have a look at it?

### CODE REVIEW

File: tests/neo.UnitTests/IO/Caching/UT_CloneCache.cs [hunk 1]
```diff
@@ -35,6 +35,7 @@ public void TestAddInternal()
 
             cloneCache.Commit();
             myDataCache[new MyKey("key1")].Should().Be(new MyValue("value1"));
+            cloneCache.Commit();
```
Comment (shargon) [inline tests/neo.UnitTests/IO/Caching/UT_CloneCache.cs:38]: It's required? maybe you should test the `GetChanges` after this

File: src/neo/IO/Caching/DataCache.cs [hunk 2]
```diff
@@ -80,12 +80,15 @@ public void Commit()
...
                         AddInternal(trackable.Key, trackable.Item);
+                        trackable.State = TrackState.None;
                         break;
                     case TrackState.Changed:
                         UpdateInternal(trackable.Key, trackable.Item);
+                        trackable.State = TrackState.None;
                         break;
                     case TrackState.Deleted:
                         DeleteInternal(trackable.Key);
+                        dictionary.Remove(trackable.Key);
```
Comment (erikzhang) [inline src/neo/IO/Caching/DataCache.cs:91]: You cannot delete items while traversing.

Comment (Qiao-Jin) [reply src/neo/IO/Caching/DataCache.cs]: Have modified correspondingly, please have a look

File: tests/neo.UnitTests/IO/Caching/UT_CloneCache.cs (hunk 1 above)
Comment (erikzhang) [reply tests/neo.UnitTests/IO/Caching/UT_CloneCache.cs:38]: It's not required. This line should be removed.

Comment (Qiao-Jin) [reply tests/neo.UnitTests/IO/Caching/UT_CloneCache.cs]: Have removed these changes

**Pull request merged**